tests/unit/s2n_self_talk_client_hello_cb_test.c

```diff
@@ -211,11 +231,38 @@ int s2n_negotiate_nonblocking_ch_cb(struct s2n_connection *conn,
     return s2n_negotiate(conn, &blocked);
 }
 
+int s2n_negotiate_nonblocking_poll(struct s2n_connection *conn,
+    struct client_hello_context *ch_ctx)
+{
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    EXPECT_NOT_NULL(conn);
+
+    EXPECT_EQUAL(ch_ctx->invoked, 0);
```

Comment:
Nit: We check for NULL to avoid seg faults when trying to dereference null pointers. In this method, you never actually dereference `conn` (you just pass the pointer through to other methods), so you could actually skip checking that it's NULL. However, you do dereference `ch_ctx` by calling `ch_ctx->invoked`, so you should check that it's not NULL.

We're less paranoid about this in tests where a seg fault is just annoying instead of a serious problem, but I thought I'd mention it since you were doing some checking here.

```suggestion
    EXPECT_NOT_NULL(conn);
    EXPECT_NOT_NULL(ch_ctx);

    EXPECT_EQUAL(ch_ctx->invoked, 0);
```